app/uk/gov/hmrc/accessibilitystatementfrontend/views/StatementPage.scala.html

```diff
@@ -20,8 +20,10 @@
 
 @this(layout: Layout)
 
-@(statement: AccessibilityStatement)(implicit request: Request[_], messages: Messages, appConfig: AppConfig)
+@(statement: AccessibilityStatement, userAction: Option[String], referrerUrl: Option[String])(
+        implicit request: Request[_], messages: Messages, appConfig: AppConfig)
 @serviceAbsoluteUrl = {https://@statement.serviceDomain@statement.serviceUrl}
+@contactFrontendUrl = {@{appConfig.contactHmrcUnauthenticatedLink + Seq(Some("?service=" + statement.contactFrontendServiceId),userAction.map(ua => "userAction=" + ua), referrerUrl.map(ru => "referrerUrl=" + ru)).flatten.mkString("&")}}
```

Comment:
I might be tempted to pull this out into a template of its own?

Reply:
Possibly also have a helper function in the package object to construct the URL

Reply:
Moved into the package object for models and added unit tets.